### PR TITLE
Flax Basics docs: Add missing `@jax.jit` to `mse`

### PR DESCRIPTION
**Pull request by rsokl**
Speeds up [linear regression example](https://flax.readthedocs.io/en/latest/notebooks/flax_basics.html#Linear-regression-with-Flax) by ~10x by jitting `mse`.

# What does this PR do?

Fixes # (issue)

## Checklist
- [ ] This PR fixes a minor issue (e.g.: typo or small bug) or improves the docs (you can dismiss the other
      checks if that's the case).
- [ ] This change is discussed in a Github issue/
      [discussion](https://github.com/google/flax/discussions) (please add a
      link).
- [ ] The documentation and docstrings adhere to the
      [documentation guidelines](https://github.com/google/flax/blob/main/docs/README.md#how-to-write-code-documentation).
- [ ] This change includes necessary high-coverage tests.
      (No quality testing = no merge!)

### CODE REVIEW

**Comment by marcvanzee** [comment]
Thank you! This is great since @cgarciae is adding testing of notebooks to our CI in #2166, so this will speed things up.

@cgarciae maybe we are missing jits in the notebooks that are currently disabled because they are too slow?

**Comment by marcvanzee** [comment]
@rsokl could you please sign the Google CLA? otherwise i can't merge this change.

**Comment by cgarciae** [comment]
@marcvanzee one of the slowest that is current excluded from testing is `annotated_mnist.ipynb` but its using `jit`. Will open an issue to review notebooks.

**Comment by rsokl** [comment]
> @rsokl could you please sign the Google CLA? otherwise i can't merge this change.

Done!

File: docs/notebooks/flax_basics.ipynb [hunk 1]
```diff
@@ -327,6 +327,7 @@
       "outputs": [],
       "source": [
         "# Same as JAX version but using model.apply().\n",
+        "@jax.jit"
```
Comment (marcvanzee) [inline docs/notebooks/flax_basics.ipynb:330]:
```suggestion
        "@jax.jit",
```

File: docs/notebooks/flax_basics.ipynb [hunk 2]
```diff
@@ -327,6 +327,7 @@
       "outputs": [],
       "source": [
         "# Same as JAX version but using model.apply().\n",
+        "@jax.jit",
```
Comment (marcvanzee) [inline docs/notebooks/flax_basics.ipynb:330]:
```suggestion
        "@jax.jit\n",
```

**Pull request merged**